### PR TITLE
Correct requiresNonNull* annotations.

### LINKED ISSUE #3149: Crashing invocation of Objects.requiresNonNullElse is permitted
The following code type-checks but should not:
```
    // This should fail typechecks, because it fails at run time.
    System.err.println((Object) requireNonNullElse(null, null));
```

A test case appears in `checker/tests/nullness/ObjectsRequireNonNullElse.java`.

### PR DESCRIPTION
**Pull request by smillst**
This is required to fixed https://github.com/typetools/checker-framework/issues/3149.

Merge https://github.com/typetools/checker-framework/pull/3175 after.

### CODE REVIEW

File: src/java.base/share/classes/java/util/Objects.java [hunk 1]
```diff
@@ -235,7 +235,7 @@ public static String toString(@GuardSatisfied @Nullable Object o, String nullDef
      * @throws NullPointerException if {@code obj} is {@code null}
      */
     @CFComment({"lock: TODO: treat like other nullness assertion methods in the Checker Framework."})
-    public static <T extends @NonNull Object> T requireNonNull(T obj) {
+    public static <T> @NonNull T requireNonNull(@NonNull T obj) {
```
Comment (smillst) [inline src/java.base/share/classes/java/util/Objects.java:238]: The previous annotations were correct, but resulted in `type.argument.type.incompatible` errors when null is passed.  These annotations produce `argument.type.incompatible` errors.  This makes it clear that the argument is the wrong type.

**Review by mernst — APPROVED** [review]

File: src/java.base/share/classes/java/util/Objects.java [hunk 2]
```diff
@@ -367,7 +367,7 @@ public static boolean nonNull(@GuardSatisfied @Nullable Object obj) {
      */
     @CFComment({"nullness: TODO: treat like other nullness assertion methods in the Checker Framework."})
     @Pure
-    public static <T extends @NonNull Object> T requireNonNull(@GuardSatisfied T obj, @GuardSatisfied Supplier<String> messageSupplier) {
+    public static <T> T requireNonNull(@GuardSatisfied @NonNull T obj, @GuardSatisfied Supplier<String> messageSupplier) {
```
Comment (wmdietl) [inline src/java.base/share/classes/java/util/Objects.java:370]: Shouldn't the return type be `@NonNull T`?

**Review by wmdietl — COMMENTED** [review]
@smillst Shouldn't there be corresponding changes to https://github.com/typetools/checker-framework/blob/master/checker/jdk/nullness/src/java/util/Objects.java#L180 ? I don't see a PR there.

File: src/java.base/share/classes/java/util/Objects.java (hunk 2 above)
Comment (smillst) [reply src/java.base/share/classes/java/util/Objects.java:370]: Yes, I've created a new pull request.  https://github.com/typetools/jdk/pull/38. I'll also update the JDK in checker-framework.

**Pull request merged**